src/main/java/service/strategy/strategyimpl/PurchaseOperation.java

```diff
@@ -0,0 +1,20 @@
+package service.strategy.strategyimpl;
+
+import dao.Storage;
+import model.FruitRecord;
+
+
+public class PurchaseOperation implements OperationHandler {
+
+    @Override
+    public void apply(FruitRecord transaction) {
+        Integer currentQuantity = Storage.storage.get(transaction.getFruit());
+        int purchaseQuantity = transaction.getQuantity();
+        if (currentQuantity == null && currentQuantity < 0) {
```

Comment:
The condition `currentQuantity == null && currentQuantity < 0` is incorrect since `currentQuantity` cannot be both `null` and less than 0 at the same time. You should check if `currentQuantity` is `null` or `currentQuantity < purchaseQuantity` before proceeding with the subtraction to ensure that there are enough fruits for the purchase.